code/modules/projectiles/guns/energy/pulse.dm

```diff
@@ -41,6 +43,8 @@
 	slot_flags = ITEM_SLOT_BELT
 	icon_state = "pulse_carbine"
 	inhand_icon_state = null
+	worn_icon_state = "gun"
+	worn_icon = 'icons/mob/clothing/belt.dmi'

```

Comment:
this is already handled by `build_worn_icon` via the slot flag, remove it.  otherwise it'll use in incorrect spite when put on armor.

Reply:
Reverted. Had the worn_icon set to make the other slots use a working icon. I'll add it to the list of missing textures.